pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -2203,6 +2228,58 @@ public boolean isSingleTenantCluster() {
     return endpointToInstance;
   }
 
+  /**
+   * Helper method to return a list of tables that exists and matches the given table name and type, or throws
+   * {@link ControllerApplicationException} if no table found.
+   * <p>When table type is <code>null</code>, try to match both OFFLINE and REALTIME table.
+   *
+   * @param tableName Table name with or without type suffix
+   * @param tableType Table type
+   * @return List of existing table names with type suffix
+   */
+  public List<String> getExistingTableNamesWithType(String tableName, @Nullable TableType tableType) {
+    List<String> tableNamesWithType = new ArrayList<>(2);
+
+    TableType tableTypeFromTableName = TableNameBuilder.getTableTypeFromTableName(tableName);
+    if (tableTypeFromTableName != null) {
+      // Table name has type suffix
+
+      if (tableType != null && tableType != tableTypeFromTableName) {
+        throw new ControllerApplicationException(LOGGER,
```

Comment:
This exception is not supposed to be thrown by PinotHelixResourceManager. You can add a Util for the rest resources for this method